apps/modernization-ui/src/apps/search/context/ColumnContextProvider.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext, useState } from 'react';
+import { DraggableLocation, DropResult } from 'react-beautiful-dnd';
+
+type ColumnProps = (source: DraggableLocation, destination: DraggableLocation) => void;
+
+type ColumnContextProps = {
+    handleDragEnd: (result: DropResult) => void;
+    displayColumns: DisplayColumn[];
+    tempColumns: DisplayColumn[];
+    saveColumns: () => void;
+    resetColumns: () => void;
+    toggleHide: (id: string) => void;
+};
+
+export type DisplayColumn = {
+    id: string;
+    name: string;
+    sortable: boolean;
+    visible: boolean;
+};
+
+const ColumnContext = React.createContext<ColumnContextProps | undefined>(undefined);
+
+const ColumnProvider: React.FC<{
+    children: React.ReactNode;
+    successCallback: () => void;
+}> = ({ children, successCallback }) => {
+    const defaultColumns = [
+        { id: 'lastNm', name: 'Legal name', sortable: false, visible: true },
+        { id: 'birthTime', name: 'Date of birth', sortable: false, visible: true },
+        { id: 'sex', name: 'Sex', sortable: false, visible: true },
+        { id: 'id', name: 'Patient ID', sortable: false, visible: true },
+        { id: 'address', name: 'Address', sortable: true, visible: true },
+        { id: 'phoneNumber', name: 'Phone', sortable: true, visible: true },
+        { id: 'names', name: 'Other names', sortable: true, visible: true },
+        { id: 'identification', name: 'ID', sortable: true, visible: true },
+        { id: 'email', name: 'Email', sortable: true, visible: true }
+    ];
```

Comment:
question:  How is this going to support search results other than Patient?

Reply:
We would need to initialize columns based on whatever table type it were.
Probably pass ColumnContext type: patient, investigation, and lab reports.